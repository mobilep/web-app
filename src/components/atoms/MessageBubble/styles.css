:root {
	--MessageBubble-background-color: rgba(224, 227, 229, .6);
	--MessageBubble-font-size: 14px;
	--MessageBubble-border-radius-small: calc(var(--borderRadius) / 2);
}

.MessageBubble {
	background-color: var(--MessageBubble-background-color);
	border-radius: var(--borderRadius);
	color: black;
	font-size: var(--MessageBubble-font-size);
	max-width: 800px;
	padding: 10px 20px 10px 10px;
	hyphens: auto;
	overflow-wrap: break-word;
}

.MessageBubble-single {
	border-top-right-radius: var(--MessageBubble-border-radius-small);
}

.MessageBubble-middle {
	border-top-right-radius: var(--MessageBubble-border-radius-small);
	border-bottom-right-radius: var(--MessageBubble-border-radius-small);
}

.MessageBubble-incoming-single {
	border-top-left-radius: var(--MessageBubble-border-radius-small);
}

.MessageBubble-incoming-middle {
	border-top-left-radius: var(--MessageBubble-border-radius-small);
	border-bottom-left-radius: var(--MessageBubble-border-radius-small);
}

.MessageBubble-partial {
	border-bottom-left-radius: 0;
	border-bottom-right-radius: 0;
	margin-bottom: 2px;
}

.MessageBubble-time {
	margin-bottom: 4px;
	font-size: 12px;
	color: var(--grey3);
}
