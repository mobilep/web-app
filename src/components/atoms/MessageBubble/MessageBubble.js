import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import './styles.css';
import SendAgo from '../SendAgo/SendAgo';

class MessageBubble extends Component {
	getClassName ({ className, incoming, singleOrLast, partial }) {
		return classNames({
			MessageBubble: true,
			[className]: true,
			'MessageBubble-single': !incoming && singleOrLast,
			'MessageBubble-incoming-single': incoming && singleOrLast,
			'MessageBubble-middle': !incoming && !singleOrLast,
			'MessageBubble-incoming-middle': incoming && !singleOrLast,
			'MessageBubble-partial': partial,
		});
	}

	render () {
		const {
			className, children, incoming, singleOrLast, timeString, partial, partialEnd,
			showSendAgo, timeStamp,
		} = this.props;

		return (
			<div className={this.getClassName({ className, incoming, singleOrLast, partial })}>
				{!partialEnd &&
					<Fragment>
						{ showSendAgo && <SendAgo date={timeStamp} /> }
						<div className='MessageBubble-time'>{timeString}</div>
					</Fragment>
				}
				{children && <div>{children}</div>}
			</div>
		);
	}
}

MessageBubble.propTypes = {
	children: PropTypes.any,
	className: PropTypes.string,
	incoming: PropTypes.bool,
	partial: PropTypes.bool,
	partialEnd: PropTypes.bool,
	showSendAgo: PropTypes.bool,
	singleOrLast: PropTypes.bool,
	timeStamp: PropTypes.number,
	timeString: PropTypes.string,
};

MessageBubble.defaultProps = {
	singleOrLast: true,
	incoming: false,
};

export default MessageBubble;
