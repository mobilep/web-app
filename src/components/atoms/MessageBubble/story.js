import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs/react';

import MessageBubble from './MessageBubble';

const messageTypeLabel = 'Message type : single/last';

const incomingLabel = 'Incoming';

storiesOf('1. Atoms / MessageBubble', module)
	.add('with text', () =>
		(<div><MessageBubble singleOrLast={boolean(messageTypeLabel, true)}
			incoming={boolean(incomingLabel, false)}>
			{text('Content',
				'Hello, my name is Kate! This is simple example of MessageBubble component! Thank you!')}
		</MessageBubble></div>)
	);
