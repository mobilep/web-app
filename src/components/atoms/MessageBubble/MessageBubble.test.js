import React from 'react';
import { shallow } from 'enzyme';

import MessageBubble from './MessageBubble';

describe('Atom - MessageBubble', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<MessageBubble>
				Text
			</MessageBubble>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render as incoming', () => {
		const wrapper = shallow(
			<MessageBubble incoming>
				Text
			</MessageBubble>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render as single or last', () => {
		const wrapper = shallow(
			<MessageBubble singleOrLast>
				Text
			</MessageBubble>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render as incoming single or last', () => {
		const wrapper = shallow(
			<MessageBubble incoming singleOrLast>
				Text
			</MessageBubble>
		);
		expect(wrapper).toMatchSnapshot();
	});

});

