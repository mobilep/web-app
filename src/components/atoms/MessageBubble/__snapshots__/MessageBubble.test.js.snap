// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - MessageBubble Should render 1`] = `
<div
  className="MessageBubble undefined MessageBubble-single"
>
  <div
    className="MessageBubble-time"
  />
  <div>
    Text
  </div>
</div>
`;

exports[`Atom - MessageBubble Should render as incoming 1`] = `
<div
  className="MessageBubble undefined MessageBubble-incoming-single"
>
  <div
    className="MessageBubble-time"
  />
  <div>
    Text
  </div>
</div>
`;

exports[`Atom - MessageBubble Should render as incoming single or last 1`] = `
<div
  className="MessageBubble undefined MessageBubble-incoming-single"
>
  <div
    className="MessageBubble-time"
  />
  <div>
    Text
  </div>
</div>
`;

exports[`Atom - MessageBubble Should render as single or last 1`] = `
<div
  className="MessageBubble undefined MessageBubble-single"
>
  <div
    className="MessageBubble-time"
  />
  <div>
    Text
  </div>
</div>
`;
